ext/drreg/drreg.c

```diff
@@ -1799,15 +2657,32 @@ drreg_event_restore_state(void *drcontext, bool restore_memory,
     }
     for (reg = DR_REG_START_GPR; reg <= DR_REG_STOP_GPR; reg++) {
         if (spilled_to[GPR_IDX(reg)] < MAX_SPILLS) {
-            reg_t val = get_spilled_value(drcontext, spilled_to[GPR_IDX(reg)]);
+            reg_t val = get_directly_spilled_value(drcontext, spilled_to[GPR_IDX(reg)]);
             LOG(drcontext, DR_LOG_ALL, 3,
                 "%s: restoring %s from slot %d from " PFX " to " PFX "\n", __FUNCTION__,
                 get_register_name(reg), spilled_to[GPR_IDX(reg)],
                 reg_get_value(reg, info->mcontext), val);
             reg_set_value(reg, info->mcontext, val);
         }
     }
+    for (reg = DR_REG_APPLICABLE_START_SIMD; reg <= DR_REG_APPLICABLE_STOP_SIMD; reg++) {
+        if (spilled_simd_to[SIMD_IDX(reg)] < MAX_SIMD_SPILLS) {
+            if (reg_is_strictly_xmm(reg)) {
+                get_indirectly_spilled_value(drcontext, reg,
+                                             spilled_simd_to[SIMD_IDX(reg)], simd_buf,
```

Comment:
Always restores zmm? From [here](https://github.com/DynamoRIO/dynamorio/pull/3893/files#r338264168) we can't tell.

Reply:
Fixed.